src/routes/photos.py

```diff
@@ -25,19 +25,41 @@
 security = HTTPBearer()
 
 
-@router.get("/{photo_id}", response_model=PhotoResponse)
-async def get_photo(photo_id: int,
-                    db: Session = Depends(get_db)):
-    photo = await repository_photos.get_photo_by_photo_id(photo_id, db)
-    if not photo:
-        raise HTTPException(status_code=404, detail="Photo not found")
-    return PhotoResponse(
-        id=photo.id,
-        url=photo.url,
-        description=photo.description,
-        created_by=photo.created_by,
-        created_at=photo.created_at,
-    )
+@router.get("/", response_model=Union[List[PhotoResponse], PhotoResponse])
+async def get_photos_by_user_id_or_all(
+        db: Session = Depends(get_db),
+        user_id: Optional[int] = None,
+        photo_id: Optional[int] = None,
+        limit: int = 10,
```

Comment:
limit: int = Query(10, gt=0, le=100),  # Ensure limit is within a valid range
    skip: int = Query(0, ge=0),  # Ensure skip is non-negative
    
    It is better to implement simple validation for a pagination (code above is for example).